src/components/Editor/index.js

```diff
@@ -224,7 +226,9 @@ class Editor extends PureComponent<Props, State> {
   }
 
   componentWillUpdate(nextProps) {
-    this.setText(nextProps);
+    if (nextProps.selectedSource !== this.props.selectedSource) {
+      this.setText(nextProps);
+    }
```

Comment:
why do we need this?

Reply:
It was resetting the text each time, we setEmptylines and other stuff even though the selected source has not changed